### ISSUE TITLE
Multiple `@Watch` support and usage of `@Watch` as methods.

### ISSUE THREAD

**Issue by supertiger1234** [misha/vue-declassify#4]
There is also a small problem with `@Watch` if you dont mind me pointing out.
I guess I'm just doing this in a bad way, but i stack watchers like this:
```js
@Watch("currentChannelID")
@Watch("currentServerID")
@Watch("currentTab")
saveLastSelected() {
  // code
}  
``` 
this gets turned into only one after running vue-declassify:
```js
watch: {
  currentChannelID: {
    handler() {
       //code
    }
```
I also reference `saveLastSelected` in my methods which doesnt work as that function doesnt exist anymore.

I'm probably asking for too much, sorry!
thanks for the quick replies though ^^

**Comment by misha** [comment]
This is also obviously a bug, I'm likely picking off only one of multiple potential watch decorators.

This is pretty straightforward to test/develop for, so I'll take a look at the same time as the other issue.

**Comment by misha** [comment]
@supertiger1234 I re-implemented watch translation.

Instead of creating an anonymous `handler` function, I first place the handler inside `methods`, then call it from the `watch` hash. This means you can also call it directly from your code. I also handled multiple `watch` options. I was only taking the first one. Thanks to the change above, both `watch` entries simply call the same `methods` entry. 

There are tests; feel free to check the commit referenced above.

This is published in version 1.0.4.

**Comment by supertiger1234** [comment]
Looks good 😊 thanks

**Comment by supertiger1234** [comment]
Seems to be working fine, I guess I'm going to close both issues :)

**Comment by supertiger1234** [comment]
Nevermind! I found another issue 😅 
```js
@Watch("emojiName")
onInput(val: string) {
  this.emojiName = val.replace(/[^A-Z0-9]+/gi, "_").trim();
}
```
everything works fine when converting but the arguments are not supplied:
```js
onInput(val: string) {
  this.emojiName = val.replace(/[^A-Z0-9]+/gi, "_").trim();
}
```

**Comment by misha** [comment]
You're totally right, it's an edge case and I didn't write a test case for it.

I guess the correct way to handle it is to make the handler have the exact same parameters as the method, since that's what the class-based syntax ultimately implies. Unfortunately you can't just do `hander: this.method` because `this` isn't well-defined without opening a function closure.

**Comment by misha** [comment]
So, I did some research about how _not_ to have to copy the parameters multiple times. In the [documentation for watch](https://vuejs.org/v2/api/#watch), apparently you can supply a string as the handler if that string is a method name, cleanly side-stepping the entire issue.

There were two drawbacks.

First, the method name is a string, so it's not type-safe. But it's in the same file and cannot be exposed as an external API, which to me is sufficiently safe for even the most strict codebases.

Second, the TypeScript definitions for the Vue 2 component model doesn't appear to support that specific feature. I was forced to emit a `@ts-ignore` just before each handler name usage to avoid generating errors. I guess the feature was added afterwards, and the TypeScript definition never got updated.

In short, it works without duplicating the parameters now (or should, according to the documentation). This is released under version 1.0.6.

Let me know if it doesn't, or if you find any other bugs. @supertiger1234 - you've single-handedly taken this project to the next level. Thank you!

**Comment by supertiger1234** [comment]
Thanks!!! I will test in a bit. I should thank YOU for fixing these bugs so quickly ^^ great work!

**Comment by supertiger1234** [comment]
Seems to be working fine but unfortunately I have a lot of typescript errors. Is there anyway I could contact you with another way?

**Comment by misha** [comment]
Closing since the feature seems to be working.
